kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt

```diff
@@ -0,0 +1,142 @@
+
+    private val regions = mutableListOf<ProximityZone>()
+    private val beaconFlow: MutableSharedFlow<BeaconScanResult> = MutableSharedFlow()
+
+    private val scanResults = mutableSetOf<BeaconScanResult>()
+
+    override fun startScanning() {
+        if (running) {
+            Log.e("AndroidEstimoteBeaconScanner", "Already scanning")
+            return
+        }
+        if (regions.isEmpty()) {
+            setRegions(listOf())
+        }
+
+        Log.d("AndroidEstimoteBeaconScanner", "Starting scanning")
+
+        observationHandler =
+            with(proximityObserver) {
+                running = true
+                flowEmitter()
+                startObserving(regions)
+            }
+    }
+
+    override fun stopScanning() {
+        if (!running) {
+            Log.e("AndroidEstimoteBeaconScanner", "Not scanning anyways")
+            return
+        }
+        observationHandler!!.stop()
+        mainJob.cancel()
+        running = false
+    }
+
+    override fun setRegions(regions: List<LokateBeacon>) {
+        if (running) {
+            Log.d("AndroidEstimoteBeaconScanner", "Already running!")
+            return
+        }
+        if (regions.isEmpty()) {
+            Log.d("AndroidEstimoteBeaconScanner", "No regions to scan")
+            return
+        }
+
+        Log.d("AndroidEstimoteBeaconScanner", "Setting regions")
+
+        this.regions.clear()
+        this.regions.addAll(
+            regions
+                .map { beacon ->
+                    ProximityZoneBuilder()
+                        .forTag(beacon.proximityUUID.uppercase() + ':' + beacon.major + ':' + beacon.minor)
+                        .inCustomRange(Double.MAX_VALUE)
+                        .onEnter { handleEnter(it) }
+                        .onExit { handleExit(it) }
+                        .build()
+                },
+        )
+    }
+
+    override fun scanResultFlow(): Flow<BeaconScanResult> {
+        return beaconFlow
+    }
+
+    private fun flowEmitter() {
+        coroutineScope.launch {
+            while (true) {
+                delay(Defaults.DEFAULT_SCAN_PERIOD)
+                Log.i("AndroidEstimoteBeaconScanner", "Beacons found: ${scanResults.map { it.minor }}")
+                scanResults.forEach {
+                    beaconFlow.emit(it.copy(seen = System.currentTimeMillis()))
+                }
+            }
+        }
+    }
+
+    private fun handleEnter(regionContext: ProximityZoneContext) {
+        val (beaconUUID, major, minor) = regionContext.tag.split(":")
+
+        scanResults.add(
+            BeaconScanResult(
+                beaconUUID = beaconUUID,
+                major = major.toInt(),
+                minor = minor.toInt(),
+                rssi = 0.0, // not relevant (handled inside EstimoteSDK)
+                txPower = 0, // not relevant (handled inside EstimoteSDK)
+                accuracy = 0.0, // definitely inside proximity range, so passes the if check in LokateSDK
```

Comment:
It looks like we cant filter this on LokateSDK while using estimote. It is better to make estimote handle the distance thing